### ISSUE TITLE
Port is not freed immediately after using server.stop()

### ISSUE THREAD

**Issue by kucharzyk-sebastian** [nodkz/mongodb-memory-server#330]
## Versions

- NodeJS: 10.15.1
- mongodb-memory-server: 6.6.3
- mongodb: 4.0.14.0
- system: Windows 10

package: mongo-memory-server

## What is the Problem?

When performing a sequence of starting, stopping and starting mongodb-memory-server on a specific port, the second start claims that the port is locked and uses random port.

## Code Example
```ts
process.env.MONGOMS_DEBUG = 1
const mms = require("mongodb-memory-server");

let mongoInstance;

async function startInMemoryMongoDB() {
  if (!mongoInstance || !mongoInstance.getInstanceInfo()) {
    mongoInstance = await mms.MongoMemoryServer.create({
      autoStart: true,
      instance: {
        port: 27017,
        ip: '127.0.0.1',
        dbName: 'dawnLocal'
      }
    });
  }
}

async function stopInMemoryMongoDB() {
  if (mongoInstance) {
    await mongoInstance.stop();
  }
}

async function main() {
  await startInMemoryMongoDB();
  await stopInMemoryMongoDB();
  await startInMemoryMongoDB();
}

main();
```

## Do you know *why* it happenes?

*no*

**Comment by kucharzyk-sebastian** [comment]
Hello,

Did you maybe have some time to take a look at that issue? Or maybe there's some other working way to restart the server immediately? 

Best regards

**Comment by hasezoey** [comment]
this may be an problem because of #287 and/or that `MongoInstance.stop` may not actually wait until everything is stopped

**Comment by hasezoey** [comment]
in-repository replication code:
https://github.com/nodkz/mongodb-memory-server/blob/4a034f069278e052ca305e409363408ff5cf53cc/packages/mongodb-memory-server-core/src/__tests__/replset-single-restart-test.ts#L38-L46

**Comment by hasezoey** [comment]
here is what is happening in more detail:
because the instance is started and stopped within less than 30 seconds, `get-port` hasnt cleared that port from the "tried ports" list, so in the next restart it will try to use another port (and not even try to re-try that port)

so in 7.0.0(`v7.0.0-beta.6`, but actually `.7`) there are some changes that can at least be an workaround:
(all mentioned commits come from #407)
- thanks to e2ae87961fa817e1d885487c9edc8ed89ac82260 the instances can be reused without any drawback
- thanks to 18c77e27cb4ed80514a3c4e893711a933c9bfe8a the port from an previous start (or custom port actually) can be forced to be used

**Comment by kucharzyk-sebastian** [comment]
Hello @hasezoey ,

Sorry for the late response. I've just had some time to test the workaround with a combination of reusing instances and forcing the port from the previous start. It seems to work great in `v7.0.0-beta.9`. Can't wait to see it working on `master`. Thanks a lot!

**Comment by hasezoey** [comment]
closing because fixed since `7.0.0-beta.9`
